cpp/src/arrow/compute/kernels/scalar_temporal_test.cc

```diff
@@ -1979,6 +2003,16 @@ TEST_F(ScalarTemporalTest, TestAssumeTimezoneNonexistent) {
                    &options_earliest);
 }
 
+TEST_F(ScalarTemporalTest, StrftimeOffsetTimezone) {
+  auto options_ymdhms = StrftimeOptions("%Y-%m-%dT%H:%M:%S");
+
+  const char* seconds = R"(["1970-01-01T01:59:00", "2021-08-18T16:12:00", null])";
+  const char* seconds_offset = R"(["1970-01-01T03:00:00", "2021-08-18T17:13:00", null])";
+
+  CheckScalarUnary("strftime", timestamp(TimeUnit::SECOND, "+01:01"), seconds, utf8(),
```

Comment:
@zilto here's an example of a minute resolution offset test.

Reply:
Thanks for checking with colleagues! Some reviewers are on holidays right now, but I think this could realistically make it into the next release (September-ish timeframe).

> Minute resolution makes sense given the format +XX:XX. Users would be responsible for handling values downstream. Values that aren't in 15min increments simply wouldn't match any "named IANA time zone"

Yeah, I think it's good as is now and hope it won't lead to footguns :D.